python/pyspark/sql/session.py

```diff
@@ -337,7 +338,7 @@ def _inferSchemaFromList(self, data):
         if type(first) is dict:
             warnings.warn("inferring schema from dict is deprecated,"
                           "please use pyspark.sql.Row instead")
-        schema = reduce(_merge_type, map(_infer_schema, data))
+        schema = reduce(_merge_type, [_infer_schema(row, names) for row in data])
```

Comment:
Not a big deal but let's use generator expression -> `(_infer_schema(row, names) for row in data)`